python-sdk/nuscenes/map_expansion/arcline_path_utils.py

```diff
@@ -0,0 +1,196 @@
+from typing import Dict, Any, List, Tuple
+import math
+
+def principal_value(angle_in_radians: float) -> float:
+    """
+    Ensures the angle is within -pi, pi.
+    :param angle_in_radians: Angle in radians.
+    :return: Scaled angle in radians.
+    """
+
+    interval_min = -math.pi
+    two_pi = 2 * math.pi
+    scaled_angle = (angle_in_radians - interval_min) % two_pi + interval_min
+    return scaled_angle
+
+def compute_segment_sign(arcline_path_3: Dict[str, Any]) -> Tuple[float, float, float]:
+    """
+    Compute the sign of an arcline_path_3 based on its shape.
+    :param arcline_path_3: arcline_patch_3 record.
+    :return: Tuple of signs for all three parts of the path. 0 if straight, -1 if right,
+        1 if left.
+    """
+    shape = arcline_path_3['shape']
+    segment_sign = [0, 0, 0]
+
+    if shape in ("LRL", "LSL", "LSR"):
+        segment_sign[0] = 1
+    else:
+        segment_sign[0] = -1
+
+    if shape == "RLR":
+        segment_sign[1] = 1
+    elif shape == "LRL":
+        segment_sign[1] = -1
+    else:
+        segment_sign[1] = 0
+
+    if shape in ("LRL", "LSL", "RSL"):
+        segment_sign[2] = 1
+    else:
+        segment_sign[2] = -1
+
+    return segment_sign
+
+def group_exponential(pose: Tuple[float, float, float],
+                      t: float) -> Tuple[float, float, float]:
+    """
+    Computes group exponential for a pose.
+    :param pose: Pose represented as tuple (x, y, yaw).
+    :param t: Exponent to raise the pose to.
+    """
+
+    theta = pose[2] * t
+    ctheta = math.cos(theta)
+    stheta = math.sin(theta)
+
+    if abs(pose[2]) < 1e-6:
+        return [pose[0] * t, pose[1] * t, theta]
+    else:
+        new_x = (pose[1] * (ctheta - 1.0) + pose[0] * stheta) / pose[2]
+        new_y = (pose[0] * (1.0 - ctheta) + pose[1] * stheta) / pose[2]
+        return [new_x, new_y, theta]
+
+def compose_right(left: Tuple[float, float, float],
+                  pose: Tuple[float, float, float]) -> Tuple[float, float, float]:
+    """
+    Composes two poses.
+    :param left: Tuple representing a pose.
+    :param pose: Tuple representing a pose.
+    :return: Pose tuple.
+    """
+
+    new_x = math.cos(left[2]) * pose[0] - math.sin(left[2]) * pose[1] + left[0]
+    new_y = math.sin(left[2]) * pose[0] + math.cos(left[2]) * pose[1] + left[1]
+    new_yaw = principal_value(left[2] + pose[2])
+
+    return [new_x, new_y, new_yaw]
+
+
+def get_breakpoint_poses(segment_sign: Tuple[float, float, float],
+                        radius: float) -> List[Tuple[float, float, float]]:
+
+    return [[1.0, 0.0, segment_sign[0] / radius],
+            [1.0, 0.0, segment_sign[1] / radius],
+            [1.0, 0.0, segment_sign[2] / radius]]
+
+def pose_at_step(arcline_path_3: Dict[str, Any],
```

Comment:
The name `step` gives connotations that we are stepping along the path (more similar to discretize). Let's rename to `s` or `l`, both of which would be slightly clearer. Typically s \in [0, 1], and `l \in [0, length]`, so maybe the latter?